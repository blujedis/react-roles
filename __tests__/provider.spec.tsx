/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import RoleManager from '../src';
import renderer from 'react-test-renderer';
import { renderHook, act } from '@testing-library/react-hooks';

const { Provider, useRoles } = RoleManager.initProvider({
  anon: 'anon',
  user: 'user',
  manager: 'manager',
  admin: 'admin',
  superadmin: 'superadmin'
});

const MyComponent = () => {

  // Simple component to test hook.
  const hook = useRoles();

  return (
    <div>
      Hello World
    </div>
  );

}

test('test role manager hook', () => {
  expect(true).toBe(true);
  const { result } = renderHook(() => useRoles())
  const hook = result.current;
  const hasRole = hook.hasRole('user', 'user');
  expect(hasRole).toBe(true);
});

test('test provider component', () => {

  const component = renderer.create(
    <Provider>
      <MyComponent />
    </Provider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});