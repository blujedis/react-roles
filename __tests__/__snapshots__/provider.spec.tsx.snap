// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test provider component 1`] = `
<div>
  Hello World
</div>
`;
